nvmeadm/tests/discovery_test.rs

```diff
@@ -31,3 +171,62 @@ fn connect_test() {
 fn disconnect_test() {
     let _ = disconnect("mynqn");
 }
+
+#[test]
+fn test_against_real_target() {
+    // Start an SPDK-based nvmf target
+    let _target = NvmfTarget::new(CONFIG_FILE, &TARGET_PORT.to_string());
+
+    // Check that we CANNOT connect to an NQN that does not exist
+    let mut explorer = DiscoveryBuilder::default()
+        .transport("tcp".to_string())
+        .traddr("127.0.0.01".to_string())
+        .trsvcid(TARGET_PORT)
+        .build()
+        .unwrap();
+
+    let pages = explorer.discover();
+    pages.iter().map(|e| println!("{:#?}", e)).for_each(drop);
+
+    let not_expected_to_connect =
+        explorer.connect("an nqn that does not exist");
+    let _ = match not_expected_to_connect {
+        Ok(_) => panic!(
+            "Problem should NOT be able to connect to {}",
+            SERVED_DISK_NQN
+        ),
+        Err(error) => error,
+    };
+
+    // Check that we CAN connect to an NQN that is served
+    let mut explorer = DiscoveryBuilder::default()
+        .transport("tcp".to_string())
+        .traddr("127.0.0.01".to_string())
+        .trsvcid(TARGET_PORT)
+        .build()
+        .unwrap();
+    let pages = explorer.discover();
+    pages.iter().map(|e| println!("{:#?}", e)).for_each(drop);
+
+    let expect_to_connect = explorer.connect(SERVED_DISK_NQN);
+    let _ = match expect_to_connect {
+        Ok(result) => result,
+        Err(error) => {
+            panic!("Problem connecting to {}: {:?}", SERVED_DISK_NQN, error)
+        }
+    };
+
+    // Check that we CAN disconnect from a served NQN
+    let mut explorer = DiscoveryBuilder::default()
+        .transport("tcp".to_string())
+        .traddr("127.0.0.01".to_string())
+        .trsvcid(TARGET_PORT)
+        .build()
+        .unwrap();
+    let pages = explorer.discover();
+    pages.iter().map(|e| println!("{:#?}", e)).for_each(drop);
```

Comment:
There were there because I found the helpful when I was working out what cargo test was doing. ('cargo test -- --nocapture' let me see them).

All gone now.